components/script/dom/htmlelement.rs

```diff
@@ -400,6 +404,28 @@ impl HTMLElementMethods for HTMLElement {
 
         rect.size.height.to_nearest_px()
     }
+
+    // https://html.spec.whatwg.org/multipage/#the-innertext-idl-attribute
+    fn InnerText(&self) -> DOMString {
+        let node = self.upcast::<Node>();
+        let window = window_from_node(node);
+        let element = self.upcast::<Element>();
+        let style = window.GetComputedStyle(element, None);
+
+        // Step 1.
+        let element_not_rendered = !node.is_in_doc() || style.GetPropertyValue(DOMString::from("display")) == "none";
```

Comment:
nit: Please avoid using `GetComputedStyle`, use `has_css_layout_box` instead. This is completely bogus otherwise.